app/configurator/components/select-dataset-step.tsx

```diff
@@ -43,17 +44,19 @@ const formatBackLink = (query: Router["query"]) => {
     subtype && subiri
       ? `${encodeURIComponent(subtype)}/${encodeURIComponent(subiri)}`
       : undefined;
-  return ["/browse", typePart, subtypePart].filter(Boolean).join("/");
+
+  const pathname = ["/browse", typePart, subtypePart].filter(Boolean).join("/");
+  return `${pathname}?${qs.stringify(queryParams)}`;
```

Comment:
Again, no need for `qs:stringify`: you can set router state with `{ pathname, query: queryParams }` instead of a string.

Reply:
I checked yesterday if this would work with Link, saw the first example with encodeURIComponent and missed the second part 🤦

Reply:
Done in https://github.com/visualize-admin/visualization-tool/pull/243/commits/b200a050567330c3e0f7134db1464ea10ad4b143